App/Models/Category.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Models;
+
+use RuntimeException;
+use App\Services\File;
+use App\Services\Json;
+use App\Interfaces\ModelCategoryInterface;
+
+class Category implements ModelCategoryInterface
+{
+    private ?array $products;
+
+    private const CATEGORY = 'category';
+
+    private const NEW_CATEGORY = 'newCategory';
+
+    public function __construct($path)
+    {
+        $encodedProducts = File::readFile($path);
+
+        $this->products = $encodedProducts ? Json::fromJson($encodedProducts) : null;
+
+        if ($this->products === null) {
+            throw new RuntimeException('Failed to open stream: No such file or directory');
+        }
+    }
+
+    public function all(): ?array
+    {
+        $categories = array_keys($this->products);
+
+        return $categories;
+    }
+    
+    public function one(string $category): ?array
+    {
+        return $this->products[$category] ?? null;
+    }
+
+    public function create(array $data): void
+    {
+        if (!isset($data[self::CATEGORY])) {
+            throw new RuntimeException('Missing `category` key in the array');
+        }
+
+        if (isset($this->products[$data[self::CATEGORY]])) {
+            throw new RuntimeException('Can\'t create, category already exists');
+        }
+
+        $this->products[$data[self::CATEGORY]] = [];
+    }
+
+    public function update(array $toUpdate): void
+    {
+        if (!isset($toUpdate[self::NEW_CATEGORY], $toUpdate[self::CATEGORY])) {
+            throw new RuntimeException('Missing `category` or `newCategory` keys in the array');
```

Comment:
Same changes can be applied in whole `Category` class

```suggestion
            throw new RuntimeException(
                sprintf(
                    'Missing `%s` or `%s` keys in the array',
                    self::CATEGORY,
                    self::NEW_CATEGORY
                )
            );
```